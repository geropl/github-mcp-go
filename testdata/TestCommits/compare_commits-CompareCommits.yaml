---
version: 2
interactions:
    - id: 0
      request:
        proto: HTTP/1.1
        proto_major: 1
        proto_minor: 1
        content_length: 0
        transfer_encoding: []
        trailer: {}
        host: api.github.com
        remote_addr: ""
        request_uri: ""
        body: ""
        form: {}
        headers:
            Accept:
                - application/vnd.github.v3+json
            User-Agent:
                - go-github/v69.2.0
            X-Github-Api-Version:
                - "2022-11-28"
        url: https://api.github.com/repos/geropl/github-mcp-go-test/compare/main...main
        method: GET
      response:
        proto: HTTP/2.0
        proto_major: 2
        proto_minor: 0
        transfer_encoding: []
        trailer: {}
        content_length: -1
        uncompressed: true
        body: '{"url":"https://api.github.com/repos/geropl/github-mcp-go-test/compare/main...main","html_url":"https://github.com/geropl/github-mcp-go-test/compare/main...main","permalink_url":"https://github.com/geropl/github-mcp-go-test/compare/geropl:4707587...geropl:4707587","diff_url":"https://github.com/geropl/github-mcp-go-test/compare/main...main.diff","patch_url":"https://github.com/geropl/github-mcp-go-test/compare/main...main.patch","base_commit":{"sha":"4707587d1c57e90aeed4c846b1b71221c73de5f3","node_id":"C_kwDOOEmhctoAKDQ3MDc1ODdkMWM1N2U5MGFlZWQ0Yzg0NmIxYjcxMjIxYzczZGU1ZjM","commit":{"author":{"name":"Gero Posmyk-Leinemann","email":"gero@gitpod.io","date":"2025-03-07T10:52:03Z"},"committer":{"name":"Gero Posmyk-Leinemann","email":"gero@gitpod.io","date":"2025-03-07T10:52:03Z"},"message":"Delete test file","tree":{"sha":"e913e2c86ad3cb8950b3a25a2d112ab5609e8928","url":"https://api.github.com/repos/geropl/github-mcp-go-test/git/trees/e913e2c86ad3cb8950b3a25a2d112ab5609e8928"},"url":"https://api.github.com/repos/geropl/github-mcp-go-test/git/commits/4707587d1c57e90aeed4c846b1b71221c73de5f3","comment_count":0,"verification":{"verified":false,"reason":"unsigned","signature":null,"payload":null,"verified_at":null}},"url":"https://api.github.com/repos/geropl/github-mcp-go-test/commits/4707587d1c57e90aeed4c846b1b71221c73de5f3","html_url":"https://github.com/geropl/github-mcp-go-test/commit/4707587d1c57e90aeed4c846b1b71221c73de5f3","comments_url":"https://api.github.com/repos/geropl/github-mcp-go-test/commits/4707587d1c57e90aeed4c846b1b71221c73de5f3/comments","author":{"login":"geropl","id":32448529,"node_id":"MDQ6VXNlcjMyNDQ4NTI5","avatar_url":"https://avatars.githubusercontent.com/u/32448529?v=4","gravatar_id":"","url":"https://api.github.com/users/geropl","html_url":"https://github.com/geropl","followers_url":"https://api.github.com/users/geropl/followers","following_url":"https://api.github.com/users/geropl/following{/other_user}","gists_url":"https://api.github.com/users/geropl/gists{/gist_id}","starred_url":"https://api.github.com/users/geropl/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/geropl/subscriptions","organizations_url":"https://api.github.com/users/geropl/orgs","repos_url":"https://api.github.com/users/geropl/repos","events_url":"https://api.github.com/users/geropl/events{/privacy}","received_events_url":"https://api.github.com/users/geropl/received_events","type":"User","user_view_type":"public","site_admin":false},"committer":{"login":"geropl","id":32448529,"node_id":"MDQ6VXNlcjMyNDQ4NTI5","avatar_url":"https://avatars.githubusercontent.com/u/32448529?v=4","gravatar_id":"","url":"https://api.github.com/users/geropl","html_url":"https://github.com/geropl","followers_url":"https://api.github.com/users/geropl/followers","following_url":"https://api.github.com/users/geropl/following{/other_user}","gists_url":"https://api.github.com/users/geropl/gists{/gist_id}","starred_url":"https://api.github.com/users/geropl/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/geropl/subscriptions","organizations_url":"https://api.github.com/users/geropl/orgs","repos_url":"https://api.github.com/users/geropl/repos","events_url":"https://api.github.com/users/geropl/events{/privacy}","received_events_url":"https://api.github.com/users/geropl/received_events","type":"User","user_view_type":"public","site_admin":false},"parents":[{"sha":"1090dd042cc6f088fa2a4794f7949d00d2a22280","url":"https://api.github.com/repos/geropl/github-mcp-go-test/commits/1090dd042cc6f088fa2a4794f7949d00d2a22280","html_url":"https://github.com/geropl/github-mcp-go-test/commit/1090dd042cc6f088fa2a4794f7949d00d2a22280"}]},"merge_base_commit":{"sha":"4707587d1c57e90aeed4c846b1b71221c73de5f3","node_id":"C_kwDOOEmhctoAKDQ3MDc1ODdkMWM1N2U5MGFlZWQ0Yzg0NmIxYjcxMjIxYzczZGU1ZjM","commit":{"author":{"name":"Gero Posmyk-Leinemann","email":"gero@gitpod.io","date":"2025-03-07T10:52:03Z"},"committer":{"name":"Gero Posmyk-Leinemann","email":"gero@gitpod.io","date":"2025-03-07T10:52:03Z"},"message":"Delete test file","tree":{"sha":"e913e2c86ad3cb8950b3a25a2d112ab5609e8928","url":"https://api.github.com/repos/geropl/github-mcp-go-test/git/trees/e913e2c86ad3cb8950b3a25a2d112ab5609e8928"},"url":"https://api.github.com/repos/geropl/github-mcp-go-test/git/commits/4707587d1c57e90aeed4c846b1b71221c73de5f3","comment_count":0,"verification":{"verified":false,"reason":"unsigned","signature":null,"payload":null,"verified_at":null}},"url":"https://api.github.com/repos/geropl/github-mcp-go-test/commits/4707587d1c57e90aeed4c846b1b71221c73de5f3","html_url":"https://github.com/geropl/github-mcp-go-test/commit/4707587d1c57e90aeed4c846b1b71221c73de5f3","comments_url":"https://api.github.com/repos/geropl/github-mcp-go-test/commits/4707587d1c57e90aeed4c846b1b71221c73de5f3/comments","author":{"login":"geropl","id":32448529,"node_id":"MDQ6VXNlcjMyNDQ4NTI5","avatar_url":"https://avatars.githubusercontent.com/u/32448529?v=4","gravatar_id":"","url":"https://api.github.com/users/geropl","html_url":"https://github.com/geropl","followers_url":"https://api.github.com/users/geropl/followers","following_url":"https://api.github.com/users/geropl/following{/other_user}","gists_url":"https://api.github.com/users/geropl/gists{/gist_id}","starred_url":"https://api.github.com/users/geropl/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/geropl/subscriptions","organizations_url":"https://api.github.com/users/geropl/orgs","repos_url":"https://api.github.com/users/geropl/repos","events_url":"https://api.github.com/users/geropl/events{/privacy}","received_events_url":"https://api.github.com/users/geropl/received_events","type":"User","user_view_type":"public","site_admin":false},"committer":{"login":"geropl","id":32448529,"node_id":"MDQ6VXNlcjMyNDQ4NTI5","avatar_url":"https://avatars.githubusercontent.com/u/32448529?v=4","gravatar_id":"","url":"https://api.github.com/users/geropl","html_url":"https://github.com/geropl","followers_url":"https://api.github.com/users/geropl/followers","following_url":"https://api.github.com/users/geropl/following{/other_user}","gists_url":"https://api.github.com/users/geropl/gists{/gist_id}","starred_url":"https://api.github.com/users/geropl/starred{/owner}{/repo}","subscriptions_url":"https://api.github.com/users/geropl/subscriptions","organizations_url":"https://api.github.com/users/geropl/orgs","repos_url":"https://api.github.com/users/geropl/repos","events_url":"https://api.github.com/users/geropl/events{/privacy}","received_events_url":"https://api.github.com/users/geropl/received_events","type":"User","user_view_type":"public","site_admin":false},"parents":[{"sha":"1090dd042cc6f088fa2a4794f7949d00d2a22280","url":"https://api.github.com/repos/geropl/github-mcp-go-test/commits/1090dd042cc6f088fa2a4794f7949d00d2a22280","html_url":"https://github.com/geropl/github-mcp-go-test/commit/1090dd042cc6f088fa2a4794f7949d00d2a22280"}]},"status":"identical","ahead_by":0,"behind_by":0,"total_commits":0,"commits":[],"files":[]}'
        headers:
            Access-Control-Allow-Origin:
                - '*'
            Access-Control-Expose-Headers:
                - ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset
            Cache-Control:
                - private, max-age=60, s-maxage=60
            Content-Security-Policy:
                - default-src 'none'
            Content-Type:
                - application/json; charset=utf-8
            Date:
                - Fri, 07 Mar 2025 13:02:37 GMT
            Etag:
                - W/"4622c2a3c7edb35bd256c1f811ec66c1db9860345265554ec9513199f6f9e6b1"
            Github-Authentication-Token-Expiration:
                - 2025-04-06 13:14:03 +0200
            Last-Modified:
                - Fri, 07 Mar 2025 10:52:03 GMT
            Referrer-Policy:
                - origin-when-cross-origin, strict-origin-when-cross-origin
            Server:
                - github.com
            Strict-Transport-Security:
                - max-age=31536000; includeSubdomains; preload
            Vary:
                - Accept, Authorization, Cookie, X-GitHub-OTP,Accept-Encoding, Accept, X-Requested-With
            X-Accepted-Github-Permissions:
                - contents=read
            X-Content-Type-Options:
                - nosniff
            X-Frame-Options:
                - deny
            X-Github-Api-Version-Selected:
                - "2022-11-28"
            X-Github-Media-Type:
                - github.v3; format=json
            X-Github-Request-Id:
                - 7498:1EA0F8:3C7CB51:3E64A08:67CAEE6C
            X-Ratelimit-Limit:
                - "5000"
            X-Ratelimit-Remaining:
                - "4943"
            X-Ratelimit-Reset:
                - "1741353443"
            X-Ratelimit-Resource:
                - core
            X-Ratelimit-Used:
                - "57"
            X-Xss-Protection:
                - "0"
        status: 200 OK
        code: 200
        duration: 267.420754ms
