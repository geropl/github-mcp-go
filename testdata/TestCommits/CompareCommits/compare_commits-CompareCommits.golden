{
  "output": "# Commit Comparison\n\n**Base Commit:** 4707587d1c57e90aeed4c846b1b71221c73de5f3  \n**Ahead By:** 0 commits  \n**Behind By:** 0 commits  \n**Status:** identical  \n**URL:** https://github.com/geropl/github-mcp-go-test/compare/main...main  \n\n## Stats\n\n**Total Commits:** 0  \n**Files Changed:** 0  \n**Additions:** 0  \n**Deletions:** 0  \n\n",
  "err": ""
}